pkg/image/importer/importer.go

```diff
@@ -299,9 +319,35 @@ func formatRepositoryError(repository *importRepository, refName string, refID s
 	return
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image, cache *imageStreamImporterCache) error {
+	bs := repo.Blobs(ctx)
+
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+
+		if layerSize, ok := cache.LayerSizeCache[layer.Name]; ok {
+			size += layerSize
+			continue
+		}
+
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
Before we start stat-ing, can we also check image manifest, schema v2 has a [Descriptor](https://github.com/openshift/origin/blob/128e499c9b5a7fb9a965234a23e40dc30f814d43/pkg/image/api/dockertypes.go#L60-L71) for [each layer](https://github.com/openshift/origin/blob/128e499c9b5a7fb9a965234a23e40dc30f814d43/pkg/image/api/dockertypes.go#L86), doesn't it contain reasonable values?